File: test-suite/golden-tests/mapped-parameterized-types/boxes.R
# As in Python: the mapping is a hint, the value is structural.
grow <- function(b) {
    if (b[[1]] == "Full") list("Full", list(b[[2]][[1]] + 1000)) else b
}

bump <- function(w) {
    w$item <- w$item + 1000
    w
}

File: test-suite/golden-tests/mapped-parameterized-types/boxes.py
# A mapped form in Python is a hint on the wire and nothing more: the value
# is the structural (constructor, fields) pair whatever the mapping says.
def grow(b):
    match b:
        case ("Full", (x,)):
            return ("Full", (x + 100,))
        case _:
            return b


def bump(w):
    return {"item": w["item"] + 100}
